File: 2022/02/02.c
#include <stdio.h>

int main() {
    FILE* file = fopen("input.txt", "r");
    int p1 = 0, p2 = 0;

    for (char o, j, y; fscanf(file, "%c%c%c%c", &o, &j, &y, &j) == 4; p2++) {
        p1 += (y - ++o) % 3 * 3 + y - 87;
        p2 += (y - 88) * 3 + (y + ++o) % 3;
    }

    printf("%d\n%d\n", p1, p2);
}

File: 2025/02/02.py
import textwrap
p1 = 0
p2 = 0
for (l, r) in [map(int, l.split("-")) for l in open("input.txt").read().split(",")]:
    for e in range(l, r+1):
        s = str(e)
        if s[:len(s)//2] == s[len(s)//2:]:
            p1 += e
        for n in range(1, len(s)//2+1):
            if len(set(textwrap.wrap(s, n))) == 1:
                p2 += e
                break

print(p1)
print(p2)
